docs/reference/evercrypt/mac/index.md

```diff
@@ -0,0 +1,94 @@
+# MAC
+
+A Message Authentication Code (MAC) provides data integrity and authenticity of a message.
+
+## HMAC
+
+An HMAC is a specific construction of a MAC that involves a cryptographic hash function (see [RFC 2104]).
+Thus, an HMAC comes in multiple instantiations.
+HACL Packages supports the following ones:
+
+* HMAC-BLAKE2b,
+* HMAC-BLAKE2s,
+* HMAC-SHA-2-256,
+* HMAC-SHA-2-384,
+* HMAC-SHA-2-512, and
+* HMAC-SHA-1.
+
+Keys must be chosen using a cryptographically strong pseudo-random generator and periodically refreshed.
+
+# HMAC
+
+## API Reference
+
+```{doxygentypedef} Spec_Hash_Definitions_hash_alg
+```
+
+All available hash functions.
```

Comment:
There's no way for a user to know what valid values are here.
Either link to the source code or add the values in here.